packages/react-native/ReactCommon/react/renderer/components/scrollview/ScrollViewEventEmitter.cpp

```diff
@@ -26,7 +26,15 @@ void ScrollViewEventEmitter::onScrollBeginDrag(
 
 void ScrollViewEventEmitter::onScrollEndDrag(
     const ScrollEvent& scrollEvent) const {
-  dispatchScrollViewEvent("scrollEndDrag", scrollEvent);
+  const auto* endDragScrollEvent =
+      dynamic_cast<const ScrollEndDragEvent*>(&scrollEvent);
```

Comment:
@cipolleschi Hi, this is the another way by `dynamic_cast` `scrollEvent` to `ScrollEndDragEvent` if we removed those changes.

Reply:
uhm... sorry for the extra back and forth. 
I tested locally and actually I prefer the previous version with the properly typed methods and without the cast.
I haven't realized that even if the `dispatchScrollViewEvent` takes an `endDragScrollEvent`, it then creates a shared pointer of type `ScrollEvent`: creating the new event pointer with explicit ScrollEvent type makes us lose the information of the actual subclass and therefore we can't serialize the event properly. :( 

I modified the imported code, nothing required to do on your side! <3